modules/replication/module.py

```diff
@@ -2,21 +2,58 @@
 
 import time
 from modules.algorithm_module import AlgorithmModule
+# from modules.enums import ReplicationEnums
+from modules.constants import (REP_STATE, R_LOG, PEND_REQS, REQ_Q,
+                               LAST_REQ, CON_FLAG, VIEW_CHANGE,
+                               REQUEST, SEQUENCE_NO, STATUS)  # X_SET, REPLY
+from copy import deepcopy
 
 
 class ReplicationModule(AlgorithmModule):
-    """Models the Replication module."""
-
-    flush = False
-    need_flush = False
-    view_changed = False
+    """Models the Replication module.
+
+    Structure of variables:
+    q (request by client): <client c, timestamp t, operation o>
+    request (accepted request): < request q, view v, sequence number seq_n>
+    rep_state = UNDEFINED
+    r_log (x_set is the set that claim to have executed/comitted request):
+        <request, x_set>
+    pend_req: <request>
+    req_q : <request, status t>
+    last_req: (last executed request for each client): <request, reply>
+
+    rep[n] (replica structure):
+        <rep_state, r_log, pend_req, req_q, last_req, con_flag, view_change>
+    """
 
     def __init__(self, id, resolver, n, f):
         """Initializes the module."""
         self.id = id
         self.resolver = resolver
         self.number_of_nodes = n
         self.number_of_byzantine = f
+        self.DEF_STATE = {REP_STATE: {},
+                          R_LOG: [],
+                          PEND_REQS: [],
+                          REQ_Q: [],
+                          LAST_REQ: [],
+                          CON_FLAG: False,
+                          VIEW_CHANGE: False}
+
+        self.flush = False
+        self.need_flush = False
+        self.seq_n = 0
+        self.rep = [
+            {REP_STATE: {},
+             R_LOG: [],
+             PEND_REQS: [],
+             REQ_Q: [],
+             LAST_REQ: [],
+             CON_FLAG: False,
+             VIEW_CHANGE: False}
+            for i in range(n)
+        ]
+        self.prim = None
```

Comment:
Might lead to problems with comparisons later, maybe use `-1` instead?

Reply:
Done!